pkg/util/fs_util.go

```diff
@@ -575,9 +579,13 @@ func CopyFile(src, dest, buildcontext string) (bool, error) {
 		return false, err
 	}
 	defer srcFile.Close()
-	uid := fi.Sys().(*syscall.Stat_t).Uid
-	gid := fi.Sys().(*syscall.Stat_t).Gid
-	return false, CreateFile(dest, srcFile, fi.Mode(), uid, gid)
+	if uid < 0 {
```

Comment:
This looks really similar to the code at https://github.com/GoogleContainerTools/kaniko/pull/962/files#diff-d36eb675aa49a7b471e3a2be77005b18R517

would it be possible to unify them?

Reply:
yes, moved to own function. thanks.